test/robot/build/package.go

```diff
@@ -150,20 +150,29 @@ func (pkg *Package) mergeTool(ctx context.Context, tool *ToolSet) {
 	for _, t := range pkg.Tool {
 		if t.Abi.SameAs(tool.Abi) {
 			// merge into existing tool
-			if tool.Interceptor != "" {
-				t.Interceptor = tool.Interceptor
+			if tool.Host.Gapir != "" {
+				t.Host.Gapir = tool.Host.Gapir
 			}
-			if tool.Gapii != "" {
-				t.Gapii = tool.Gapii
+			if tool.Host.Gapis != "" {
+				t.Host.Gapis = tool.Host.Gapis
 			}
-			if tool.Gapir != "" {
-				t.Gapir = tool.Gapir
+			if tool.Host.Gapit != "" {
+				t.Host.Gapit = tool.Host.Gapit
 			}
-			if tool.Gapis != "" {
-				t.Gapis = tool.Gapis
-			}
-			if tool.Gapit != "" {
-				t.Gapit = tool.Gapit
+			for _, android := range tool.Android {
+				if android.GapidApk != "" {
+					merged := false
+					for _, a := range t.Android {
```

Comment:
Maybe add a `func (*ToolSet) FindAndroidToolSet(*device.ABI) *AndroidToolSet` helper method that returns `nil` if the ABI cannot be found?